File: templates/javaee-app/application-services/src/main/java/org/sgodden/tom/services/impl/CustomerOrderServiceImpl.java
package org.sgodden.tom.services.impl;

import org.sgodden.tom.model.CustomerOrderFactory;
import org.sgodden.tom.model.CustomerOrderRepository;
import org.sgodden.tom.model.ICustomerOrder;
import org.sgodden.tom.model.services.CustomerOrderService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

@Service
@Transactional
public class CustomerOrderServiceImpl implements CustomerOrderService {

    @Autowired
    private CustomerOrderRepository repository;

    @Autowired
    private CustomerOrderFactory factory;

    @Override
    public ICustomerOrder create() {
        return factory.create();
    }

    @Override
    public Long persist(ICustomerOrder customerOrder) {
        repository.persist(customerOrder);
        return (Long) customerOrder.getId();
    }

    @Override
    public void remove(Long id) {
        repository.remove(repository.findById(id));
    }

    @Override
    public List<ICustomerOrder> findAll() {
        return repository.findAll();
    }

    @Override
    public void merge(ICustomerOrder order) {
        repository.merge(order);
    }

    @Override
    public ICustomerOrder findById(Long id) {
        return repository.findById(id);
    }

}

File: templates/scala-mongodb/application-services/src/main/scala/org/sgodden/tom/services/customerorder/impl/CustomerOrderServiceImpl.scala
package org.sgodden.tom.services.customerorder.impl

import org.sgodden.tom.services.customerorder.CustomerOrderService
import org.sgodden.tom.model.{ICustomerOrder, CustomerOrderFactory, CustomerOrderRepository}
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
class CustomerOrderServiceImpl extends CustomerOrderService {

  @Autowired private var repository: CustomerOrderRepository = null
  @Autowired private var factory: CustomerOrderFactory = null

  override def create: ICustomerOrder = factory.create

  override def persist(customerOrder: ICustomerOrder) = {
    repository.persist(customerOrder)
    customerOrder.getId
  }

  override def remove(id: String) = repository.remove(repository.findById(id))

  override def findAll: List[ICustomerOrder] = repository.findAll

  override def merge(order: ICustomerOrder) =  repository merge order

  override def findById(id: String): ICustomerOrder = repository.findById(id)

}
